newIDE/app/src/UI/Carousel.js

```diff
@@ -0,0 +1,356 @@
+            description: '',
+            thumbnail: '',
+          })
+          .map((item, index) => ({ ...item, id: `skeleton${index}` }));
+
+  const windowSize = useResponsiveWindowWidth();
+  const imageHeight = referenceSizesByWindowSize.imageHeight[windowSize];
+  const arrowWidth = referenceSizesByWindowSize.arrowWidth[windowSize];
+  const cellWidth = (16 / 9) * imageHeight;
+  const widthUnit = cellWidth + cellSpacing;
+
+  const cellHeight =
+    imageHeight + (displayItemTitles ? titleHeight + spacerSize : 0);
+
+  const renderImage = React.useCallback(
+    (item: ThumbnailType | SkeletonThumbnail): React.Node => (
+      <img
+        src={item.thumbnailUrl}
+        style={{
+          ...styles.image,
+          height: imageHeight,
+          minHeight: imageHeight,
+          width: cellWidth,
+        }}
+        alt={item.title}
+        title={item.title}
+      />
+    ),
+    [cellWidth, imageHeight]
+  );
+
+  const renderThumbnail = React.useCallback(
+    (item: ThumbnailType | SkeletonThumbnail): ?React.Node => {
+      if (!item.skeleton && !item.link && !item.thumbnailUrl) return null;
+      if (item.link) {
+        return (
+          <a href={item.link} target="_blank">
+            {renderImage(item)}
+          </a>
+        );
+      }
+      if (item.thumbnailUrl) {
+        return renderImage(item);
+      }
+      if (item.skeleton) {
+        return (
+          <Skeleton variant="rect" height={imageHeight} width={cellWidth} />
+        );
+      }
+    },
+    [renderImage, cellWidth, imageHeight]
+  );
+
+  const renderItemTitle = React.useCallback(
+    (item: ThumbnailType | SkeletonThumbnail): ?React.Node => {
+      if (!displayItemTitles) return null;
+      return (
+        <>
+          <Spacer />
+          {item.title ? (
+            <div style={{ ...styles.itemTitleContainer, width: cellWidth }}>
+              <Text noMargin style={styles.itemTitle}>
+                {item.title}
+              </Text>
+            </div>
+          ) : (
+            <Skeleton
+              variant="rect"
+              height={titleHeight}
+              width={(cellWidth / 3) * (1 + 2 * Math.random())} // Make rectangles of different lengths so that the UI feels more "alive".
+            />
+          )}
+        </>
+      );
+    },
+    [cellWidth, titleHeight]
+  );
+
+  const renderBrowseAllButton = (): React.Node => {
+    const flatButtonProps: $Shape<FlatButtonProps> = { onClick: () => {} };
+    if (!onBrowseAllClick && browseAllLink) {
+      flatButtonProps.onClick = () => {
+        Window.openExternalURL(browseAllLink);
+      };
+      flatButtonProps.target = '_blank';
+    } else if (onBrowseAllClick) {
+      flatButtonProps.onClick = onBrowseAllClick;
+    } else {
+      flatButtonProps.onClick = () => {};
+    }
+
+    return (
+      <FlatButton
+        {...flatButtonProps}
+        label={<Trans>Browse all</Trans>}
+        tabIndex={tabIndexOffset + itemsToDisplay.length}
```

Comment:
I really think the tabIndex are not necessary, tabIndex="0" should work on the tiles and are not needed for buttons. Much like z-index, this will create "fights" between components one day or the other.